File: pinyougou-manager-web/src/main/webapp/js/angularjs/controller/brandController.js
app.controller('brandController',function ($scope,brandService,$controller) {

    //继承 baseController  两个控制器 共享$scope
    $controller('baseController',{$scope:$scope})

    //读取列表数据绑定到表单当中
    $scope.findPage = function (page, rows) {
        $http.get('../brand/findPage.do?page='+page+'&rows='+rows).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;//更新总记录数
        })
    }
    //定义搜索对象
    $scope.searchEntity={}
    $scope.search = function(page, rows){
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        )

    }




    //	添加数据
    $scope.save = function () {
        //将更新和添加放在一个方法里
        var methodName = 'add';//方法名称
        if ($scope.entity.id != null) { // 如果查询的ID存在 则变为更新
            methodName = "update";
        }

        brandService.save(methodName,$scope.entity).success(function (response) {
            if (response.success){
                //重新加载
                $scope.reloadList();
            }else {
                alert(response.message)
            }

        });

    }
    //实体查询
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    }

    //批量删除
    $scope.dele = function () {
        //获取复选框的值
        brandService.dele($scope.selectIds).success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }

            }
        )

    }

})

File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import org.apache.log4j.Logger;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.lang.reflect.MalformedParameterizedTypeException;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    private Logger logger = Logger.getLogger(BrandController.class);

    @Reference
    BrandService brandService;
    @RequestMapping("/search")
    public PageResult findPage(@RequestBody TbBrand brand,int page, int rows) {
        return brandService.findPage(brand,page,rows);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {

            brandService.add(brand);
            return new Result(true,"增加成功");
        }catch (Exception e){
            logger.error("BrandController--add 添加品牌时出错"+e.getMessage());
            return new Result(false,"增加失败");
        }

    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            logger.error("BrandController--update 报错："+e.getMessage());
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    @RequestMapping("/delete")
    public Result delete(Long []ids) {
        try {
            brandService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            logger.error("BrandController -- delete 失败："+e.getMessage());
            return new Result(false, "删除失败");
        }
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }
}
